docs/rfcs/0012-introduce-ui-store.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: page
+title: Introduce a Store for UI components
+permalink: /rfc/0012-introduce-ui-store
+---
+
+* Start date: 2024-01-29
+* RFC PR: [#5353](https://github.com/mozilla-mobile/firefox-android/pull/5353)
+
+## Summary
+
+In most applications of the `Store`, it is preferable to have reducers perform work on the main thread. Having actions reduced immediately at the point of dispatch, simplifies the reasoning a developer would need to go through for most UI-based work that happens on the main thread.
```

Comment:
> Having actions reduced immediately at the point of dispatch

I do agree with this, but I think this still misses the piece of observation. Even if we reduce immediately, if we rely on `Flow` collection to observe the results at the UI level then we still don't have guarantees about immediacy.

Reply:
Addressed [here](https://github.com/mozilla-mobile/firefox-android/pull/5353#issuecomment-1974174872).